tests/solana_tests/primitives.rs

```diff
@@ -599,6 +604,196 @@ fn truncate_uint(n: &mut U256, width: usize) {
     }
 }
 
+#[test]
+
+fn test_mul_within_range_signed() {
+    let mut rng = rand::thread_rng();
+    for width in (72..=256).step_by(8) {
+        let src = r#"
+        contract test {
+         
+            function mul(intN a, intN b) public returns (intN) {
+                return a * b;
+            }
+        }"#
+        .replace("intN", &format!("int{}", width));
+
+        let mut contract = build_solidity(&src);
+
+        // Signed N bits will fit rthe range: 2^(N-1) -1 +/-. Here we generate a random number within this range and multiply it by -1, 1 or 0.
+        let first_operand_rand = rng.gen_bigint(width - 1).sub(1_u32);
+
+        let side = vec![-1, 0, 1];
+        // -1,1 or 0
+        let second_op = BigInt::from(*side.choose(&mut rng).unwrap() as i32);
+
+        contract.constructor("test", &[]);
+        let return_value = contract.function(
+            "mul",
+            &[
+                ethabi::Token::Int(bigint_to_eth(
+                    &first_operand_rand,
+                    width.try_into().unwrap(),
+                )),
+                ethabi::Token::Int(bigint_to_eth(&second_op, width.try_into().unwrap())),
+            ],
+            &[],
+            None,
+        );
+
+        let res = first_operand_rand.mul(second_op);
+        assert_eq!(
+            return_value,
+            vec![ethabi::Token::Int(bigint_to_eth(
+                &res,
+                width.try_into().unwrap(),
+            ))]
+        );
+    }
+}
+
+#[test]
+
+fn test_mul_within_range() {
+    let mut rng = rand::thread_rng();
+    for width in (72..=256).step_by(8) {
+        let src = r#"
+        contract test {
+            function mul(uintN a, uintN b) public returns (uintN) { 
+                return a * b;
+            }
+        }"#
+        .replace("intN", &format!("int{}", width));
+
+        let mut contract = build_solidity(&src);
+        contract.constructor("test", &[]);
+        for _ in 0..10 {
+            // Max number to fit unsigned N bits is 0..(2^N)-1
+            let limit = BigUint::from(2_u32).pow(width).sub(1_u32);
+
+            // Generate a random number within the the range 0..2^N -1
+            let first_operand_rand = rng.gen_biguint(width.into());
+
+            // Calculate a number that when multiplied by first_operand_rand, the result will not overflow N bits .(The result of this division will cast the float result to int result, therefore lowering the result. The result of multiplication will never overflow)
```

Comment:
```suggestion
            // Calculate a number that when multiplied by first_operand_rand, the result will not overflow N bits (the result of this division will cast the float result to int result, therefore lowering it. The result of multiplication will never overflow).
```
`result` was repeated.